### PR TITLE
Add support for the Resource Estimation compilation stage

### PR DESCRIPTION
**Pull request by gwwatkin**
Need it to work on https://github.com/latticesurgery-com/lattice-surgery-compiler/issues/227. Shouldn't affect other things.

Hopefully when https://github.com/latticesurgery-com/lattice-surgery-compiler/pull/220 goes in we can switch from this parsing to using JSON directly

### CODE REVIEW

File: src/components/sections/LatticeView.tsx [hunk 1]
```diff
@@ -88,52 +87,26 @@ const LatticeView = ({ compilationResult }: LatticeViewProps): JSX.Element => {
...
 
-            {showCompilationText ? (
+            {showCompilationText && (
                 <Flex gap={10} justifyContent={"center"} pt={0} pb={4} flexWrap={"wrap"}>
-                    <Box
-                        className="box-hover"
-                        textAlign="center"
-                        borderWidth="4px"
-                        borderRadius="xl"
-                        boxShadow={"xl"}
```
Comment (Keelando) [inline src/components/sections/LatticeView.tsx:98]: ChakraUI doesn't seem to mind if the class attributes are wrapped in curly braces or just quotes. I'm not sure if our linters will be able to make it consistent, but I think if we can, we should remove the curly braces where possible to be consistent

Comment (Keelando) [reply src/components/sections/LatticeView.tsx:98]: This doesn't apply to this PR, but just in general

File: src/components/sections/LatticeView.tsx [hunk 2]
```diff
@@ -88,52 +87,26 @@ const LatticeView = ({ compilationResult }: LatticeViewProps): JSX.Element => {
...
                             boxShadow={"xl"}
-                            p={5}
+                            key={idx}
+                            p={4}
                         >
-                            <Text className="line-1">Litinski Transform</Text>
-                            <Box pt="5" pb="2">
-                                <pre>{circuit_after_litinski}</pre>
+                            <Text className="line-1">{stage.name}</Text>
+                            <Box pt={1}>
```
Comment (Keelando) [inline src/components/sections/LatticeView.tsx:103]: pt={1} and pt="1" are equivalent I believe, and I think quotes make more sense unless they are necessary

**Review by Keelando — APPROVED** [review]

File: src/components/sections/LatticeView.tsx (hunk 2 above)
Comment (gwwatkin) [reply src/components/sections/LatticeView.tsx:103]: Whatever prettier sets :shrug:  ... I didn't change that line myself

File: src/components/sections/LatticeView.tsx (hunk 1 above)
Comment (gwwatkin) [reply src/components/sections/LatticeView.tsx:98]: I trust in the linters for these things

File: src/parseCompilationText.ts [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
         const circuit_after_litinski_split = compilation_text_split[3].split(":")
         const circuit_after_litinski = circuit_after_litinski_split[1].slice(1)
-        compilationTextJson.circuit_after_litinski = circuit_after_litinski
+        stages.push(new CompilationStage("Litinski Transform", circuit_after_litinski))
+    }
+
+    if (compilation_text.includes("resource")) {
+        const circuit_after_resource_split = compilation_text_split[4].split(":\n")
+        const circuit_after_resource = circuit_after_resource_split[1]
+        stages.push(new CompilationStage("Resource Estimation", circuit_after_resource))
```
[the remark this answers is not in the document]
Comment (gwwatkin) [reply src/parseCompilationText.ts:26]: Yes, all this ugly parsing will be replaced by a beautiful:
```ts
stages = responseJson.stages.map( stage =>
    new CompilationStage(stage.name, stage.body)
)
```

[the remark this answers is not in the document]
Comment (gwwatkin) [reply src/parseCompilationText.ts:26]: or maybe even better ... we won't need the `CompilationStage` at all ... just make it into an interface and add it to the response type and voila

**Comment by gwwatkin** [comment]
@Keelando do you feel strongly about the linting things or can I just merge

**Comment by Keelando** [comment]
Go right ahead, seems like something we'd circle back to. I was more curious about the parsing file.

**Comment by gwwatkin** [comment]
@Keelando, yes! don't even think we'll need any kind of object building ... just tunrn the stage class into an interface, add it to `CompilationResultSuccess` in `apiResponses.ts` and have it match what the compilation pipeline spits out

**Pull request merged**